src/org/zaproxy/zap/extension/ascanrulesAlpha/SQLInjectionMsSQL.java

```diff
@@ -0,0 +1,319 @@
+					return;
+				} else {
+					//phew.  the second time came in within the limits. use the later timing details as the base time for the checks.
+					originalTimeUsed = originalTimeUsed2;
+					originalTimeStarted = originalTimeStarted2;
+				}
+			}
+			//end of timing baseline check
+
+			int countTimeBasedRequests = 0;
+
+			if ( this.debugEnabled ) log.debug("Scanning URL ["+ getBaseMsg().getRequestHeader().getMethod()+ "] ["+ getBaseMsg().getRequestHeader().getURI() + "], field ["+ paramName + "] with value ["+paramValue+"] for MsSQL Injection");
+
+			//Check for time based SQL Injection, using MsSQL specific syntax
+			for (int timeBasedSQLindex = 0;
+					timeBasedSQLindex < SQL_MSSQL_TIME_REPLACEMENTS.length && doTimeBased && countTimeBasedRequests < doTimeMaxRequests;
+					timeBasedSQLindex ++) {
+				HttpMessage msgAttack = getNewMsg();
+				String newTimeBasedInjectionValue =
+						SQL_MSSQL_TIME_REPLACEMENTS[timeBasedSQLindex].
+								replace (ORIG_VALUE_TOKEN, paramValue).
+								replace(SLEEP_TOKEN, getSleepToken(sleep));
+				setParameter(msgAttack, paramName, newTimeBasedInjectionValue);
+				//send it.
+				long modifiedTimeStarted = System.currentTimeMillis();
+				try {
+					sendAndReceive(msgAttack, false); //do not follow redirects
+					countTimeBasedRequests++;
+					}
+				catch (java.net.SocketTimeoutException e) {
+					//this is to be expected, if we start sending slow queries to the database.  ignore it in this case.. and just get the time.
+					if ( this.debugEnabled ) log.debug("The time check query timed out on ["+msgTimeBaseline.getRequestHeader().getMethod()+"] URL ["+msgTimeBaseline.getRequestHeader().getURI().getURI()+"] on field: ["+paramName+"]");
+				}
+				long modifiedTimeUsed = System.currentTimeMillis() - modifiedTimeStarted;
+
+				if ( this.debugEnabled ) log.debug ("Time Based SQL Injection test: ["+ newTimeBasedInjectionValue + "] on field: ["+paramName+"] with value ["+newTimeBasedInjectionValue+"] took "+ modifiedTimeUsed + "ms, where the original took "+ originalTimeUsed + "ms");
+				//add some small leeway on the time, since adding a 5 (by default) second delay in the SQL query will not cause the request
+				//to take a full 5 (by default) seconds longer to run than the original..
+				if (modifiedTimeUsed >= (originalTimeUsed + sleep*1000 - 200)) {
+					//takes more than 5(by default) extra seconds => likely time based SQL injection. Raise it
+					String extraInfo = Constant.messages.getString("ascanalpha.sqlinjection.mssql.alert.timebased.extrainfo", newTimeBasedInjectionValue, modifiedTimeUsed, paramValue, originalTimeUsed);
+
+					//raise the alert
+					bingo(Alert.RISK_HIGH, Alert.CONFIDENCE_MEDIUM, getName(), getDescription(),
+							getBaseMsg().getRequestHeader().getURI().getURI(), //url
```

Comment:
- The URI should be passed encoded (at least that's the core/default behaviour).
 - The comment could be removed.